rct229/rulesets/ashrae9012019/section11/section11rule7.py

```diff
@@ -0,0 +1,181 @@
+from rct229.utils.jsonpath_utils import find_all
+
+SERVICE_WATER_HEATING_SPACE = SchemaEnums.schema_enums[
+    "ServiceWaterHeatingSpaceOptions2019ASHRAE901"
+]
+
+
+class Section11Rule7(RuleDefinitionListIndexedBase):
+    """Rule 7 of ASHRAE 90.1-2019 Appendix G Section 11 (Service Water Heating)"""
+
+    def __init__(self):
+        super(Section11Rule7, self).__init__(
+            rmds_used=produce_ruleset_model_description(
+                USER=False, BASELINE_0=True, PROPOSED=True
+            ),
+            each_rule=Section11Rule7.RMDRule(),
+            index_rmd=BASELINE_0,
+            id="11-7",
+            description="Except in buildings that will have no service water heating loads, the service water heating system type in the baseline building design shall be as specified in Table G3.1.1-2 for each building area type in the proposed design.",
+            ruleset_section_title="Service Water Heating",
+            standard_section="Table G3.1 #11, baseline column, a & b",
+            is_primary_rule=True,
+            list_path="ruleset_model_descriptions[0]",
+            required_fields={"$": ["calendar"], "$.calendar": ["is_leap_year"]},
+            data_items={"is_leap_year": (BASELINE_0, "calendar/is_leap_year")},
+        )
+
+    class RMDRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section11Rule7.RMDRule, self).__init__(
+                rmds_used=produce_ruleset_model_description(
+                    USER=False,
+                    BASELINE_0=True,
+                    PROPOSED=True,
+                ),
+                index_rmd=BASELINE_0,
+                each_rule=Section11Rule7.RMDRule.SWHBATRule(),
+            )
+
+        def create_data(self, context, data):
+            rmd_p = context.PROPOSED
+            rmd_b = context.BASELINE_0
+
+            service_water_heating_uses_p = {
+                swh_use["id"]: swh_use.get("use", 0.0)
+                for swh_use in find_all(
+                    "$.buildings[*].building_segments[*].zones[*].spaces[*].service_water_heating_uses[*]",
+                    rmd_p,
+                )
+            }
+
+            swh_equip_type_b = {
+                swh_equip_id: get_swh_equipment_type(rmd_b, swh_equip_id)
+                for swh_equip_id in find_all(
+                    "$.service_water_heating_equipment[*].id", rmd_b
+                )
+            }
+
+            return {
+                "service_water_heating_uses_p": service_water_heating_uses_p,
+                "swh_equip_type_b": swh_equip_type_b,
+            }
+
+        def create_context_list(self, context, data=None):
+            rmd_b = context.BASELINE_0
+            rmd_p = context.PROPOSED
+            is_leap_year_b = data["is_leap_year"]
+
+            swh_bats_and_uses_b = get_swh_components_associated_with_each_swh_bat(
+                rmd_b, is_leap_year_b
+            )
+            swh_bats_and_uses_p = get_swh_bats_and_swh_use(rmd_p)
```

Comment:
The two parameter names are confusing - they contain different data structure while the names are the same.

Reply:
Good catch! Addressed.